devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,238 @@
+/**
+ * 	Aeotec Doorbell 6
+ *
+ * 	Copyright 2019
+ *
+ * 	Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * 	in compliance with the License. You may obtain a copy of the License at:
+ *
+ * 		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * 	Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ * 	on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ * 	for the specific language governing permissions and limitations under the License.
+ *
+ */
+metadata {
+    definition(name: "Aeotec Doorbell 6", namespace: "SmartThings", author: "SmartThings", mnmn: "SmartThings") {
+        capability "Actuator"
+        capability "Health Check"
+        capability "Tamper Alert"
+        capability "Alarm"
+
+        fingerprint mfr: "0371", prod: "0003", model: "00A2", deviceJoinName: "Aeotec Doorbell 1"
+    }
+
+    tiles {
+        multiAttributeTile(name: "alarm", type: "generic", width: 6, height: 4) {
+            tileAttribute("device.alarm", key: "PRIMARY_CONTROL") {
+                attributeState "off", label: 'off', action: 'alarm.siren', icon: "st.alarm.alarm.alarm", backgroundColor: "#ffffff"
+                attributeState "both", label: 'ring!', action: 'alarm.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#0e7507"
+            }
+        }
+        standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
+            state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+        }
+        standardTile("reset", "device.reset", height: 2, width: 2, inactiveLabel: false, decoration: "flat") {
+            state "default", label: "reset", action: "reset"
+        }
+        valueTile("tamper", "device.tamper", height: 2, width: 2, decoration: "flat") {
+            state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+            state "detected", label: 'tampered', backgroundColor: "#ffffff"
+        }
+
+        main "alarm"
+        details(["alarm", "off", "tamper"])
+    }
+}
+
+def installed() {
+    initialize()
+    def numberOfSounds = 8
```

Comment:
you have `def numberOfSounds = 8` in about three places, that should be pulled out to a separate function